class.geocoder.php

```diff
@@ -198,4 +185,89 @@ private function dawa_geocode ( $address ) {
             'lng' => $json[0]->adgangsadresse->adgangspunkt->koordinater[0]
         );
     }
+
+    /** 
+     * gets a single location's coordinates from the cached locations
+     */
+    public function get_cache($address_key, $plain_address) {
+        /** 
+         * @since 3.4.0
+         * using 'leaflet_geocoder_get_cache', 
+         * you can return any value that is not identical to the address_key to avoid using get_transient
+         */
+        $filtered = apply_filters( 'leaflet_geocoder_get_cache', $address_key, $plain_address );
+
+        if ($filtered === $address_key) {
+            return get_transient( $address_key );
+        }
+
+        return $filtered;
+    }
+
+    /** 
+     * gets the array of saved locations and updates an individual location
+     */
+    public function set_cache($key, $value) {
+        /** 
+         * @since 3.4.0
+         * using 'leaflet_geocoder_set_cache', 
+         * you can return any falsy value to omit the set_transient
+         */
+        if (apply_filters('leaflet_geocoder_set_cache', $key, $value)) {
+            // get user-defined expiry (maybe this should be an admin option?)
+            $expiry = apply_filters('leaflet_geocoder_expiry', null);
+
+            if ($expiry === null) {
+                // stagger caches between 200-400 days to prevent all caches expiring on the same day
+                $stagger = random_int(200, 400);
+                $expiry = DAY_IN_SECONDS * $stagger;
+            }
+
+            set_transient( $key, $value, $expiry );
+        }
+    }
+
+    /**
+     * Appends an address to a list of addresses in the db, for cleanup
+     */
+    public function update_caches( $address ) {
+        /** 
+         * @since 3.4.0
+         * using 'leaflet_geocoder_update_caches', 
+         * you can return any falsy value to omit the set_transient
+         */
+        if (apply_filters('leaflet_geocoder_update_caches', $address)) {
+            $locations = get_transient( self::$locations_key );
+            
+            if (!$locations) {
+                $locations = array();
+            }
+            
+            array_push( $locations, $address );
+            
+            // set to 25 year expiry since we never really want it to expire
+            // but omitting expiry causes it to autoload
+            set_transient( self::$locations_key, $locations, YEAR_IN_SECONDS * 25 );
```

Comment:
the transient which holds all address keys is 25 years to avoid expiry and avoid autoload